src/predict.rs

```diff
@@ -469,11 +470,13 @@ pub trait Intra: Dim {
     bit_depth: usize
   ) {
     if alpha == 0 { return; }
+    assert!(32 >= Self::W);
+    assert!(ac.len() >= 32 * Self::H);
+    assert!(stride >= Self::W);
+    assert!(output.len() >= stride * Self::H);
```

Comment:
If you move them in the function body, does the compiler elide the checks for you even if you use the normal iterators/array indexing? (It should)

Reply:
I'll try that for the next change. The CI failures highlight that these assertions are imprecise. I'll update to permit CfL on the bottom row in positions other than the left-most column.